api/src/main/resources/org/openmrs/module/mirebalaisreports/sql/mexicoSsaMeds.sql

```diff
@@ -0,0 +1,37 @@
+SELECT
+    d.name "Medicamento",
+    SUM(disp_quant.value_numeric) "Cantidad"
+FROM obs disp_quant
+         JOIN obs disp_drug
+              ON disp_drug.obs_group_id = disp_quant.obs_group_id
+                  AND disp_drug.concept_id = (
+                      SELECT concept_id
+                      FROM concept_reference_map
+                      WHERE concept_reference_term_id = (
+                          SELECT crt.concept_reference_term_id
+                          FROM concept_reference_term crt
+                                   JOIN concept_reference_source crs
+                                        ON crt.concept_source_id = crs.concept_source_id
+                          WHERE crt.code LIKE "medication orders"
+                            AND crs.name LIKE "PIH"
+                      )
+                  )
+         INNER JOIN drug d
+                    ON disp_drug.value_drug = d.drug_id
+WHERE disp_quant.concept_id = (
+    SELECT concept_id
+    FROM concept_reference_map
+    WHERE concept_reference_term_id = (
+        SELECT crt.concept_reference_term_id
+        FROM concept_reference_term crt
+                 JOIN concept_reference_source crs
+                      ON crt.concept_source_id = crs.concept_source_id
+        WHERE crt.code LIKE "1443"
+          AND crs.name LIKE "CIEL"
+    )
+)
+  AND date(disp_quant.obs_datetime) >= :startDate
+  AND date(disp_quant.obs_datetime) <= :endDate
+  AND d.name LIKE "SSA: %"
```

Comment:
This line is the only diifference -- the other says `LIKE "CES: %"`

Reply:
Fine with me.  Main comment would be to try to avoid duplicating code where possible, as it leaves 2 independent places to remember to fix or adjust when changes are needed.  In this case, you could approach it in one of 2 ways:

1. Have a SqlDataSetDefinition that takes in an additional parameter called "nameFilter", which would be used in the LIKE constraint at the end.  Then, you would use this in the two reports, in the first by setting "CES: %" as this parameter value, and in this by setting "SSA: %" as the parameter value.  This way, you only maintain one version of the query and can re-use that same dataset for lots of different medication queries.

2. Have 2 separate reports that create independent datasets, but put the query into a single stored procedure that both of them call, where the stored procedure takes in the name filter string as an input parameter.  Then, the SQL for this report is reduced to "call dispensed_medication_and_quantity('SSA: %');"

Reply:
Is it practicable to do (1) in a way that leaves open the possibility of exporting to two different Excel files? I've been using the ReportManager system, which seems to have a 1:1 relation between datasets and reports.

Reply:
Yes, that's easy.  You just need to construct the dataset outside of the report manager (eg. in a "DefinitionLibrary" or even just in a utility class/method, and then pull the dataset in from here in your reportmanager.

Reply:
Sweet. I'm going to push this through because of the urgency, and file another ticket to refactor.

Reply:
https://pihemr.atlassian.net/browse/MEX-420